File: frontend/src/services/leadService.js
const API_URL = "https://devlab-backend-4d8f.onrender.com/api/leads";

// Create Lead
export const createLead = async (leadData) => {
    const response = await fetch(API_URL, {
        method: "POST",
        headers: {
            "Content-Type": "application/x-www-form-urlencoded",
        },
        body: new URLSearchParams(leadData),
    });

    return await response.json();
};

// Get All Leads
export const getAllLeads = async () => {
    const response = await fetch(API_URL);

    return await response.json();
};

// Get Lead By ID
export const getLeadById = async (id) => {
    const response = await fetch(`${API_URL}?id=${id}`);

    return await response.json();
};

// Update Lead Status
export const updateLeadStatus = async (id, status) => {
    const response = await fetch(
        `${API_URL}?id=${id}&status=${encodeURIComponent(status)}`,
        {
            method: "PUT",
        }
    );

    return await response.json();
};

// Delete Lead
export const deleteLead = async (id) => {
    const response = await fetch(`${API_URL}?id=${id}`, {
        method: "DELETE",
    });

    return await response.json();
};

File: backend/src/main/java/com/devlab/service/LeadService.java
package com.devlab.service;

import com.devlab.dao.LeadDAO;
import com.devlab.model.Lead;

import java.util.List;

public class LeadService {

    private LeadDAO leadDAO;

    public LeadService() {
        leadDAO = new LeadDAO();
    }

    // CREATE
    public boolean addLead(Lead lead) {
        return leadDAO.addLead(lead);
    }

    // READ ALL
    public List<Lead> getAllLeads() {
        return leadDAO.getAllLeads();
    }

    // READ ONE
    public Lead getLeadById(int id) {
        return leadDAO.getLeadById(id);
    }

    // UPDATE
    public boolean updateLead(Lead lead) {
        return leadDAO.updateLead(lead);
    }

    // DELETE
    public boolean deleteLead(int id) {
        return leadDAO.deleteLead(id);
    }
}